multines.js

```diff
@@ -49,57 +50,60 @@ function register (server, options, next) {
         deliverMap[path] = buildDeliver(socket, topic)
       }
 
-      wrapSubscribe(socket, path, params, (err) => {
-        if (err) {
-          return next(err)
-        }
-
-        mq.on(topic, deliverMap[path], next)
+      await wrapSubscribe(socket, path, params)
+      await new Promise((resolve, reject) => {
+        mq.on(topic, deliverMap[path], (err) => {
+          if (err) return reject(err)
+          resolve()
+        })
```

Comment:
can you use util.promisify instead?

Reply:
@mcollina I've tried that but it complains about `mqemitter` accessing a variable on `this` (ie: `Cannot read property '_matcher' of undefined`)

Reply:
https://github.com/mcollina/mqemitter/blob/master/mqemitter.js#L76

Reply:
@mcollina sorry, found what you meant. it's ugly but it does the trick

```util.promisify(mq.on.bind(mq))(topic, deliverMap[path])```